lib/browser/tag/mkdom.js

```diff
@@ -54,24 +53,16 @@ function specialTags(el, tmpl, tagName) {
   Replace the yield tag from any tag template with the innerHTML of the
   original tag in the page
 */
-function replaceYield(tmpl, html) {
+function replaceYield(tmpl, src) {
   // do nothing if no yield
   if (!reHasYield.test(tmpl)) return tmpl
 
-  // be careful with #1343 - string on the source having `$1`
-  var src = {}
-
-  html = html && html.replace(reYieldSrc, function (_, ref, text) {
```

Comment:
Hi everyone,

It appears the original logic for replaceYield removed any <yield to> tags if no <yield from> tags were found in the markup. For example, if I had a tag whose HTML was simply a <yield/>, but the content being yielded included a tag which uses multi-slot transclusion, then replaceYield would remove the inner <yield to> tags, causing nothing to be transcluded.

```html
<my-yielding-tag>
  <my-transcluding-tag>
    <yield to="slot-1">content 1</yield>
    <yield to="slot-2">content 2</yield>
  </my-transcluding-tag>
</my-yielding-tag>
```

I fixed this locally by wrapping lines 64-67 in a conditional: `if (reYieldDest.test(tmpl)) { ... }` but it looks like this PR addresses the issue in another (better) way.

@uk-taniyama Do you think this would fix the issue I explained above? If so, can you fix the merge conflicts so the PR can continue? Alternatively, let me know if there's any way I can help.

Thanks!